lib/puppet/provider/service/smf.rb

```diff
@@ -74,6 +76,47 @@ def startcmd
     end
   end
 
+  # Wait for the service to transition into the specified state before returning.
+  # This is necessary due to the asynchronous nature of SMF services.
+  # desired_state should be online, offline, disabled, or uninitialized.
+  # See PUP-5474 for long-term solution to this issue.
+  def wait(desired_state)
```

Comment:
Could this just be `*desired_state`?  Would mean we don't need to flatten or convert to an array and the one call that does need multiple states can just do `wait('offline', 'disabled', 'uninitialized')` which is a little cleaner looking imo.